server/src/application/survey.ts

```diff
@@ -2059,9 +2058,30 @@ export async function storeFile({
   surveyId: number;
   organizationId: string;
 }) {
+  // Check if duplicate file exists
+
   const fileString = `\\x${buffer.toString('hex')}`;
-  const fileUrl = `${organizationId}/${path.join('/')}/${name}`;
-  const row = await getDb().oneOrNone<{ path: string[]; name: string }>(
+  const splittedFileNameArray = name.split('.');
+  const extension = splittedFileNameArray.pop();
+  let fileUrl = `${organizationId}/${path.join('/')}/${splittedFileNameArray.join('.')}.${extension}`;
+  console.log(fileUrl);
+
+  for (let i = 1; i < 100; i++) {
```

Comment:
[nit] Maybe a small comment here for this `for` loop, and why does it repeat 100 times? It might look a bit confusing to run into otherwise

Reply:
If one survey is used as a template the 100 limit could get exceeded and then files would be overriden. 

Heres and alternative to drop the loop:

```suggestion
   const searchFileUrl = `${organizationId}/${path.join('/')}/${splittedFileNameArray.join('.')}%.${extension}`;

  const { count } = await getDb().one<{ count: number }>(
    `
        SELECT count(id) FROM data.files WHERE url LIKE $1;
      `,
    [searchFileUrl],
  );

  const fileUrl = `${organizationId}/${path.join('/')}/${splittedFileNameArray.join('.')}${count > 0 ? `-${count}` : ''}.${extension}`;
```

Reply:
If one survey is used as a template the 100 limit could get exceeded and then files would be overriden. 

Here's and alternative to drop the loop:

NOTE: This is not intended to be committed directly here.

```suggestion
   const searchFileUrl = `${organizationId}/${path.join('/')}/${splittedFileNameArray.join('.')}%.${extension}`;

  const { count } = await getDb().one<{ count: number }>(
    `SELECT count(id) FROM data.files WHERE url LIKE $1;`,
    [searchFileUrl],
  );

  const fileUrl = `${organizationId}/${path.join('/')}/${splittedFileNameArray.join('.')}${count > 0 ? `-${count}` : ''}.${extension}`;
```